apollo-router-core/src/plugin_utils/structures/queryplanner_response.rs

```diff
@@ -0,0 +1,36 @@
+use super::{from_names_and_values, CompatRequest};
+use crate::{Context, Object, QueryPlan};
+use futures::executor::block_on;
+use serde_json_bytes::Value;
+use std::sync::Arc;
+use typed_builder::TypedBuilder;
+
+#[derive(Default, Clone, TypedBuilder)]
+#[builder(field_defaults(default, setter(strip_option)))]
+pub struct QueryPlannerResponse {
+    #[builder(default, setter(!strip_option, transform = |extensions: Vec<(&str, Value)>| Some(from_names_and_values(extensions))))]
+    extensions: Option<Object>,
```

Comment:
could you explain what that `extensions` field is doing? I  see it is passed around everywhere, but what is its purpose? Is it inserted in `Error::extensions`? if the `QueryPlannerResponse` already contains a `Context`, why not put it there right away?

Reply:
related to https://github.com/apollographql/router/pull/484#discussion_r811320027